pkg/envoy/proxy.go

```diff
@@ -128,10 +128,15 @@ func (p Proxy) GetAnnouncementsChannel() chan announcements.Announcement {
 }
 
 // NewProxy creates a new instance of an Envoy proxy connected to the xDS servers.
-func NewProxy(cn certificate.CommonName, ip net.Addr) *Proxy {
+func NewProxy(certCommonName certificate.CommonName, certSerialNumber certificate.SerialNumber, ip net.Addr) *Proxy {
 	return &Proxy{
-		CommonName: cn,
-		Addr:       ip,
+		CommonName:   certCommonName,
+		SerialNumber: certSerialNumber,
```

Comment:
What does SerialNumber for a proxy even mean? Can we please abstract this as a part of an attribute called UID or something, where UID = SerialNumber + ...

Reply:
It is the SerialNumber of the xDS certificate.
The SerialNumber of the xDS certificate is different than the Pod UID and is different than the Envoy UID.

We can rename these struct fields to be more specific - `XDSCertificateSerialNumber` and `XDSCertificateCommonName` - would that make it clear?

Reply:
Why is the Envoy UID not a part of this struct? I don't see a reason for the SerialNumber to be a part of this struct as its own attribute. Logically it doesn't look right to expose an attribute of a certificate on the Proxy object. 

I understand what SerialNumber means in this context, but someone not familiar with the code is going to wonder what a proxy's SerialNumber used for. If the SerialNumber is going to be used just for the purpose of logging as an identifier for a proxy, it might be better to expose this via a function `proxy.GetProxyID()` which the callers can use to log the identifer as opposed to logging the SerialNumber of a certificate which looks pretty odd to reason.

Reply:
@shashankram The Envoy UID is not readily available at gRPC connect. This comes later during the xDS hand-shake.
What we have at T0, when Envoy connects to xDS, is the mTLS certificate. That certificate is unique to the Envoy, and unique to the Pod. That certificate has a SerialNumber and a CN.  Keeping these in the Proxy struct makes sense because these are unique to the Proxy.
We could and should add Envoy Node ID, but that would be eventually available.

Reply:
I didn't mean to refer to the UID as the one that is a part of the handshake, but rather an ID we can use as a part of the logs.